core/src/main/java/org/infinispan/container/impl/EntryFactoryImpl.java

```diff
@@ -101,25 +179,38 @@ public void wrapEntryForWriting(InvocationContext ctx, Object key, int segment,
             log.tracef("Updated context entry %s -> %s", contextEntry, mvccEntry);
       } else {
          // Not in the context yet.
-         CacheEntry cacheEntry = getFromContainerForWrite(key, segment, isOwner);
-         if (cacheEntry == null) {
-            return;
-         }
-         MVCCEntry mvccEntry = createWrappedEntry(key, cacheEntry);
-         if (cacheEntry.isNull()) {
-            mvccEntry.setCreated(true);
-         }
-         if (isRead) {
-            mvccEntry.setRead();
+         InternalCacheEntry ice = getFromContainer(key, segment);
+         if (isOwner) {
+            if (ice == null) {
+               addWriteEntryToContext(ctx, NullCacheEntry.getInstance(), key, isRead);
+            } else {
+               CompletionStage<Boolean> expiredStage = handlePossibleExpiredEntry(ice, true);
+               if (expiredStage != null) {
+                  returnedStage = expiredStage.thenApply(expired -> {
```

Comment:
no checking if the `expiredStage` is completed as in `wrapEntryForReading()`?

Reply:
Fixed.